instances.py

```diff
@@ -0,0 +1,17 @@
+import praw
+import pdb
+import re
+
+
+alreadyreplied=[] 		#Contains the submission ID of the submissions to which the bot has already replied to
+
+def get_jokes(subreddit1,subreddit):
+	for submission in subreddit.hot(limit=10):
+		if submission.id not in alreadyreplied:
```

Comment:
This has complexity `O(n)` because `alreadyreplied` is a list. If you make `alreadyreplied` a [set](https://docs.python.org/3/tutorial/datastructures.html#sets) or a dict, the complexity will reduce to `O(1)` because they're hash tables. Also, alreadyreplied shouldn't be global, I guess.